src/org/opendatakit/common/cli/Cli.java

```diff
@@ -95,6 +97,12 @@ public void run(String[] args) {
       operations.forEach(operation -> {
         if (cli.hasOption(operation.param.shortCode)) {
           checkForMissingParams(cli, operation.requiredParams);
+          if (operation.param.shortCode.equals("e")) {
```

Comment:
`Cli` must not know anything about the operation it runs. These checks can't be done here. I think it's fine to let `LocalDate.parse()` throw a `DateTimeParseException`. We could catch that on the Param's mapper and transform it to a `BriefcaseException` that will get gracefully caught by `Cli` afterwards.

The only problematic thing is that the mapper is getting quite big. Maybe it's time we create a factory for `Param<LocalDate>` like this:

```java
class Param<T> {
public static Param<LocalDate> localDate(String shortCode, String longCode, String description) {
    return Param.arg(
        shortCode,
        longCode,
        description + " (yyyy-MM-dd or yyyy/MM/dd)",
        dateAsText -> {
          try {
            return LocalDate.parse(dateAsText.replaceAll("/", "-"));
          } catch (DateTimeParseException e) {
            throw new BriefcaseException("Invalid date format");
          }
        }
    );
  }
}
```

Reply:
I will submit the changes by tonight :)